src/darkleaf/di/core.clj

```diff
@@ -829,3 +830,29 @@
             (p/demolish factory obj)
             (after-demolish! {:key key :object obj})
             nil))))))
+
+
+(defn- inspect-middleware []
+  (fn [registry]
+    (fn [key]
+      (let [factory       (registry key)
+            declared-deps (p/dependencies factory)
+            info          (cond-> {:key key}
+                            (seq declared-deps)
+                            (assoc :dependencies declared-deps))]
+        (reify p/Factory
+          (dependencies [_]
+            declared-deps)
+          (build [_ deps]
+            (into [info]
+                  (comp
+                   (mapcat val)
+                   (distinct))
+                  deps))
+          (demolish [_ obj]))))))
+
+(defn inspect [key & middlewares]
```

Comment:
Inspects system dependency graph.

   Returns vector of component  like
   ```clojure
   [{:key `root :dependencies {`foo :required `bar :optional}}]
   ```

  Use it instead of `start` to inspect ...

  See the tests for use cases.

Reply:
```
Collects and returns a vector of keys along with their dependencies, which is useful for inspecting enabled components and services. Evaluates all registries with middlewares applied.

Expects the same arguments as `start and returns a vector of maps e.g.:
[{:key `root :dependencies {`foo :required `bar :optional}}]
```